examples/map/map_segment.py

```diff
@@ -0,0 +1,118 @@
+"""
+=======================================================
+Segmenting a Map based on transformation of coordinates
+=======================================================
+
+A demostration creating a segment of a particular map from
+transformed coordinates.
+"""
+import numpy as np
+import matplotlib.pyplot as plt
+from reproject import reproject_interp
+
+import astropy.units as u
+from astropy.wcs import WCS
+from astropy.coordinates import SkyCoord
+
+import sunpy.map
+from sunpy.coordinates import Helioprojective, NorthOffsetFrame, RotatedSunFrame, transform_with_sun_center
+from sunpy.data.sample import AIA_171_ROLL_IMAGE
+
+###############################################################################
+# We start with the sample data.
+smap = sunpy.map.Map(AIA_171_ROLL_IMAGE)
+
+###############################################################################
+# A utility function gives us access to the helioprojective coordinate of each
+# pixel. From this we then transform the coordinates to the required frame.
+all_hpc = sunpy.map.all_coordinates_from_map(smap)
+all_hgs = all_hpc.transform_to("heliographic_stonyhurst")
+
+###############################################################################
+# Let's then segment the data based on coordinates and convert them to pixel
+# index. We convert the segmented pixel values to 0.
```

Comment:
I think this comment needs to explain a little more about what this line is doing. It's creating a boolean mask where True is the masked out segment (I think).

Reply:
Yeah, I shall word it properly